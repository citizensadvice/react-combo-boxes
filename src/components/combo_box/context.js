import { createContext } from 'react';
import { visuallyHiddenClassName } from '../../constants/visually_hidden_class_name';

export const Context = createContext({
  visuallyHiddenClassName,
  dispatch: null,
});
