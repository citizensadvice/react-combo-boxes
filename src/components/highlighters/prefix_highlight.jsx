import React from 'react';
import { HighlightValue } from '../highlight_value';
import { prefixHighlighter } from '../../highlighters/prefix_highlighter';

export function PrefixHighlight(props) {
  return (
    <HighlightValue
      highlight={prefixHighlighter}
      {...props}
    />
  );
}
