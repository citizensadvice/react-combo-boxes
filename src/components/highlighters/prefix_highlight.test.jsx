import React from 'react';
import { render } from '@testing-library/react';
import { PrefixHighlight } from './prefix_highlight';
import { Context } from '../../context';

function TestHighlight({ children, inverse, value, ...props }) {
  return (
    <Context.Provider value={{ ...props, props: { value } }}>
      <PrefixHighlight inverse={inverse}>
        {children}
      </PrefixHighlight>
    </Context.Provider>
  );
}

it('does not highlight with no children', () => {
  const { container } = render((
    <TestHighlight search="bar" />
  ));

  expect(container).toContainHTML('<div></div>');
});

it('does not highlight with no search children', () => {
  const { container } = render((
    <TestHighlight search={null}>
      foo
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>foo</div>');
});

it('does not highlight no match', () => {
  const { container } = render((
    <TestHighlight search="bar">
      foo bar
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>foo bar</div>');
});

it('highlights the first prefix', () => {
  const { container } = render((
    <TestHighlight search="bar">
      bar foo bar
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div><mark>bar</mark> foo bar</div>');
});

it('highlights existing value', () => {
  const { container } = render((
    <TestHighlight value={{ label: 'bar' }}>
      bar foo bar
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div><mark>bar</mark> foo bar</div>');
});

it('inverses the highlight', () => {
  const { container } = render((
    <TestHighlight search="bar" inverse>
      bar foo bar
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>bar<mark> foo bar</mark></div>');
});
