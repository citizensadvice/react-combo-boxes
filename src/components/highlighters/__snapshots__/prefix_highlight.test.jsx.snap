// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlights existing value 1`] = `
<div>
  <span
    class="sr-only"
  >
    bar foo bar
  </span>
  <span
    aria-hidden="true"
  >
    <mark>
      bar
    </mark>
     foo bar
  </span>
</div>
`;

exports[`highlights the first prefix 1`] = `
<div>
  <span
    class="sr-only"
  >
    bar foo bar
  </span>
  <span
    aria-hidden="true"
  >
    <mark>
      bar
    </mark>
     foo bar
  </span>
</div>
`;

exports[`inverses the highlight 1`] = `
<div>
  <span
    class="sr-only"
  >
    bar foo bar
  </span>
  <span
    aria-hidden="true"
  >
    bar
    <mark>
       foo bar
    </mark>
  </span>
</div>
`;
